segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageManager.java

```diff
@@ -0,0 +1,1375 @@
+import io.pravega.segmentstore.storage.metadata.MetadataTransaction;
+import io.pravega.segmentstore.storage.metadata.SegmentMetadata;
+import io.pravega.segmentstore.storage.metadata.StorageMetadataAlreadyExistsException;
+import io.pravega.segmentstore.storage.metadata.StorageMetadataVersionMismatchException;
+import io.pravega.segmentstore.storage.metadata.StorageMetadataWritesFencedOutException;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implements segment storage using {@link ChunkStorageProvider} and {@link ChunkMetadataStore}.
+ * The metadata about the segments is stored in metadataStore using two types of records {@link SegmentMetadata} and {@link ChunkMetadata}.
+ * Any changes to layout must be made inside a {@link MetadataTransaction} which will atomically change the records upon {@link MetadataTransaction#commit()}
+ */
+@Slf4j
+public class ChunkStorageManager implements Storage {
+    /**
+     * Default SegmentRollingPolicy to use.
+     */
+    @Getter
+    private SegmentRollingPolicy defaultRollingPolicy;
+
+    /**
+     * Metadata store containing all storage data.
+     */
+    @Getter
+    private ChunkMetadataStore metadataStore;
+
+    /**
+     * Underlying {@link ChunkStorageProvider} to use to read and write data.
+     */
+    @Getter
+    private ChunkStorageProvider chunkStorage;
+
+    /**
+     * Storage executor object.
+     */
+    private Executor executor;
+
+    /**
+     * Tracks whether this instance is closed or not.
+     */
+    private final AtomicBoolean closed;
+
+    /**
+     * Current epoch of the {@link Storage} instance.
+     */
+    @Getter
+    private long epoch;
+
+    /**
+     * Id of the current Container.
+     */
+    @Getter
+    private int containerId;
+
+    /**
+     * {@link SystemJournal} that logs all changes to system segment layout so that they can be are used during system bootstrap.
+     */
+    @Getter
+    private SystemJournal systemJournal;
+
+    /**
+     * Size of chunk in bytes above which it is no longer considered a small object.
+     * For small source objects, concatWithAppend is used instead of using concatNatively.
+     */
+    @Getter
+    @Setter
+    private long minSizeLimitForNativeConcat;
+
+
+    /**
+     * Size of chunk in bytes above which it is no longer considered for concat.
+     */
+    @Getter
+    @Setter
```

Comment:
Remove the setter. Pass this via config.

Reply:
fixed.